services/autorust/codegen/src/codegen.rs

```diff
@@ -240,15 +240,11 @@ impl CodeGen {
             if &nm.to_string() != property_name {
                 serde_attrs.push(quote! { rename = #property_name });
             }
-            if property.schema.read_only == Some(true) {
-                serde_attrs.push(quote! { skip_serializing });
```

Comment:
It is specific to the client side. I want to be able to do round-trip testing and I want to use the models from the server-side. One reason for read-only is so that the client doesn't send a bunch of the data back to the server. It is an optimization that isn't needed right now. Another design would be to generate a function that could clear the read only values, may be `.clear_read_only()`.

Reply:
It's a good optimization that I don't think we do in other language SDKs right now. I imagine the fear is that, for PATCH operations (could be implemented as a PUT as well), missing data handled incorrectly could result in data loss. @pakrym might have more insight into why we still send read-only data back, or if it's just an optimization we haven't yet done.

Reply:
>@pakrym might have more insight into why we still send read-only data back

Do we? We shouldn't be.

Reply:
Good to know we don't for .NET. Thanks.

Reply:
And, that said, if it hasn't been a detected problem yet this is probably a safe optimization. Ideally, if we did find a problem it should be a service-side fix since Azure has guidelines on patching resources.